### ISSUE TITLE
IPMasquerade not work with IPv6

### ISSUE THREAD

**Issue by nmdnm** [systemd/systemd#14909]
**systemd version the issue has been seen with**
> systemd 244 (244.3-1-arch)

**Used distribution**
 > Arch Linux

**Expected behaviour you didn't see**
 > Add masquerade to iptables for all networks from config file

**Unexpected behaviour you saw**
 > Add masquerade to iptables for all ipv4 networks from config file

**Steps to reproduce the problem**
Create /etc/systemd/network/wg.network:
```
[Match]
Name = wg0

[Network]
Address = fd00::1/48
Address = 10.0.0.1/24
IPForward=True
IPMasquerade=True
```
Create /etc/systemd/network/wg.netdev:
```
[NetDev]
Name = wg0
Kind = wireguard

[WireGuard]
ListenPort = 51820
PrivateKey = PRIVATE_KEY

[WireGuardPeer]
PublicKey = PUBLIC_KEY
AllowedIPs = fd00::2/48,10.0.0.2/24
```
`$ sudo systemctl restart systemd-networkd`

Result:
```
$ sudo iptables -S -t nat | grep MASQUERADE
-A POSTROUTING -s 10.0.0.0/24 -j MASQUERADE
$ sudo ip6tables -S -t nat | grep MASQUERADE
$
```

**Comment by ssahani** [comment]
IPv6  is not currently supported . RFE

**Comment by ssahani** [comment]
reconsidering this I am not sure how to proceed about it . 
```
https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/8/html/configuring_and_managing_networking/getting-started-with-nftables_configuring-and-managing-networking

32.2. When to use firewalld, nftables, or iptables

The following is a brief overview in which scenario you should use one of the following utilities:

firewalld: Use the firewalld utility to configure a firewall on workstations. The utility is easy to use and covers the typical use cases for this scenario.
nftables: Use the nftables utility to set up complex firewalls, such as for a whole network.
iptables: The iptables utility is deprecated in Red Hat Enterprise Linux 8. Use instead nftables.
```

Iptables is deprecated.  Should we support Nftables and replace iptables with Nftables . I am now sure how to proceed.  @poettering @keszybz @yuwata ?

**Comment by mbiebl** [comment]
@ssahani #13307 ?

**Comment by ssahani** [comment]
> @ssahani #13307 ?

@mbiebl thanks but I am pretty sure should we integrate the libs or we can write on our own. libnftable uses the netlink libs to talk to kernel and I think integrating the libnftable will be suffice and expanding out netlink lib can handle this.

**Comment by datafoo** [comment]
I found this issue after failing to make IPv6 work (proper result on https://ipv6-test.com/) with Wireguard and networkd. @vikaig: did you find an alternative way to make IPv6 work?

**Comment by gdonval** [comment]
As far as I know, NAT should not be used with IPv6. It does not mean people should be prevented from using it, don't get me wrong, it just means that the normal idiomatic way to solve the problem here is by assigning global routable IPv6 addresses to peers.

**Comment by yuwata** [comment]
Implemented by #18007.
